### PR TITLE
Image handling for real time listener and post reporting

### PR DESCRIPTION
**Pull request by godiabolo**
Here is code we modified to be able to handle image sending for the realtime mode with the listener and also a new parameter to handle images when they are not in the root directory of the project. 
Hope it can help other projects.

### CODE REVIEW

File: robotframework_reportportal/listener.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ def log_message(self, message):
         """
         msg = self._build_msg_struct(message)
         logger.debug('ReportPortal - Log Message: {0}'.format(message))
+        msg.handle_img_attachment()
```
Comment (iivanou) [inline robotframework_reportportal/listener.py:71]: It should not be here.

File: robotframework_reportportal/model.py [hunk 2]
```diff
@@ -227,3 +229,29 @@ def __init__(self, message):
     def __repr__(self):
         """Return string representation of the object."""
         return self.message
+
+    def handle_img_attachment(self):
```
Comment (iivanou) [inline robotframework_reportportal/model.py:233]: It should not be here. Moreover, the implementation can be done much simpler. Look into the pathlib library.

File: robotframework_reportportal/result_visitor.py [hunk 3]
```diff
@@ -141,8 +144,9 @@ def start_message(self, msg):
             try:
                 m = self.parse_message(message['message'])
                 message["message"] = m[0]
-                listener.log_message_with_image(message, m[1])
-            except (AttributeError, IOError):
+                image_filename=os.path.join(self.logroot, m[1])
```
Comment (iivanou) [inline robotframework_reportportal/result_visitor.py:147]: Put the logic(handle_img_attachment) for finding attachments right here.

**Review by iivanou — CHANGES_REQUESTED** [review]

File: robotframework_reportportal/model.py (hunk 2 above)
Comment (HardNorth) [inline robotframework_reportportal/model.py:233]: Looks too complex, requires tests.

**Review by HardNorth — CHANGES_REQUESTED** [review]

File: robotframework_reportportal/listener.py (hunk 1 above)
Comment (deepio-oc) [reply robotframework_reportportal/listener.py:71]: For real time use case, it needs to be here right? I dont see result_visitor.py getting invoked when robot is launched with --listener argument

**Pull request closed without merging**